cloudigrade/api/tests/clouds/azure/tasks/maintenance/test_repopulate_azure_instance_definitions.py

```diff
@@ -0,0 +1,114 @@
+from django.test import TestCase
+
+from api.clouds.azure import tasks
+from api.models import InstanceDefinition
+
+_faker = faker.Faker()
+
+# Token for this string since some of the paths in "patch" uses are very long.
+_MAINTENANCE = "api.clouds.azure.tasks.maintenance"
+
+
+class RepopulateAzureInstanceMappingTest(TestCase):
+    """Celery task 'repopulate_azure_instance_mapping' test cases."""
+
+    def setUp(self):
+        """Set up common variables for tests."""
+        self.standard_a5_definition = Mock()
+        self.standard_a5_definition.name = "Standard_A5"
+        self.standard_a5_definition.resource_type = "virtualMachines"
+        vcpu_capability = Mock()
+        vcpu_capability.name = "vCPUs"
+        vcpu_capability.value = 2
+        memory_capability = Mock()
+        memory_capability.name = "MemoryGB"
+        memory_capability.value = 14
+        capabilities_list = [vcpu_capability, memory_capability]
+        self.standard_a5_definition.capabilities = capabilities_list
+        self.good_json = json.dumps(
+            self.standard_a5_definition, default=lambda x: x.__dict__
+        )
+
+        self.bad_standard_a5_definition = Mock()
+        self.bad_standard_a5_definition.name = "Standard_A5"
+        self.bad_standard_a5_definition.resource_type = "virtualMachines"
+        vcpu_capability = Mock()
+        vcpu_capability.name = "vCPUs"
+        vcpu_capability.value = 1.21
+        memory_capability = Mock()
+        memory_capability.name = "MemoryGB"
+        memory_capability.value = 41
+        capabilities_list = [vcpu_capability, memory_capability]
+        self.bad_standard_a5_definition.capabilities = capabilities_list
+        self.bad_json = json.dumps(
+            self.bad_standard_a5_definition, default=lambda x: x.__dict__
+        )
+
+        self.garbage_definition = Mock()
+        self.garbage_definition.name = "RandomAzureObject"
+        self.garbage_definition.resource_type = "notVirtualMachines"
+
+    @patch("api.util.InstanceDefinition.objects.get_or_create")
+    @patch(f"{_MAINTENANCE}.ComputeManagementClient")
+    def test_repopulate_azure_instance_mapping(self, mock_compute, mock_db_create):
+        """Test that repopulate_azure_instance_mapping creates db objects."""
+        mock_instance_def = Mock()
+        mock_instance_def.instance_type.return_value = "Standard_A5"
+        mock_db_create.return_value = (mock_instance_def, True)
+
+        resource_skus = mock_compute.return_value.resource_skus
+        resource_skus.list = MagicMock(
+            return_value=[self.standard_a5_definition, self.garbage_definition]
+        )
+
+        tasks.repopulate_azure_instance_mapping()
+
+        mock_compute.assert_called_once()
+        resource_skus.list.assert_called_once()
+        mock_db_create.assert_called_with(
+            instance_type="Standard_A5",
+            cloud_type="azure",
+            defaults={"memory": 14, "vcpu": 2, "json_definition": self.good_json},
```

Comment:
Since we now have higher decimal precision on how we store the memory and vcpu values, how would you feel about changing this or adding another test to exercise that precision? Might be unnecessary; just a passing thought.